eknrenderer/eknrenderer/eknr-boxed-set-info.h

```diff
@@ -0,0 +1,32 @@
+/* Copyright 2018 Endless Mobile, Inc. */
+
+#pragma once
+
+#include <glib-object.h>
+
+G_BEGIN_DECLS
+
+typedef struct _EknrBoxedSetInfo {
+  guint    ref_count;
+  gboolean featured;
+  char    *ekn_id;
+  char    *title;
+  GStrv    child_tags;
+  GStrv    tags;
+} EknrBoxedSetInfo;
```

Comment:
I wouldn't want to use the actual content object model here since part of the point of this PR was to break the dependency on that.

That said, would it be better if this was an actual object? I decided to use a boxed type initially because it was less boilerplate to write for something that was basically there for marshalling data between the caller and the renderer.

Reply:
Hmm, it makes no sense from the point of view of the SDK as a platform if this library can't take EkncContentObjectModels, because then we have to marshal between EkncContentObjectModel and EknrBoxedSetInfo in our code. Maybe in that case we're better off duplicating the code...

Reply:
The issue is that I need to use this library as a separately included module in the companion app and I can't have an ekncontent dependency there. It makes more sense to just pass the primitives around, no?

Reply:
Well, I still have to review the whole thing to get the full context, but if we have to resort to acrobatics to avoid duplicating the code... maybe it would just be better to duplicate the code, and share only the Mustache templates or something like that?

Reply:
@smspillaz My opinion here is that we should try to minimize as much as we can the type proliferation. Perhaps instead of passing the whole table around, we can have API as part of the `render()` call that replaces what we'd need the table for.
Without having given it *too* much thought, for instance, we could have a "tag determination" function that replaces the hash table lookup, and a to-JSON serialization API.

Reply:
> Without having given it too much thought, for instance, we could have a "tag determination" function that replaces the hash table lookup, and a to-JSON serialization API.

I mean, sure, but when we have to go reimplement the exact same logic in the companion app again. I'd prefer to avoid having to do that....

Reply:
>  but if we have to resort to acrobatics to avoid duplicating the code... 

I'd really prefer to avoid duplicating the code. The reality is that you'll still need to populate the now-copied Ekncontent structs from something else (though perhaps if ekncontent is a separate library that means that at least from e-k-l we have ABI compatibiltiy across compiled libraries and can pass those structs directly from there ...).

But in any event, if the choice is between duplicating the code and having to copy data around or something else, I'd prefer a "something else", with perhaps the compromise that we define some structs to make passing around what we need a little nicer.

Reply:
This way just requires duplication of different code, though — the code to marshal the data into the `EknrBoxedSetInfo`.

Reply:
On the plus side of `EknrBoxedSetInfo` - you're only exposing things to the renderer library that it actually needs. However, I can see your point. What would your thoughts be on linking to ekncontent-as-a-library (eg, the companion app links to both as libraries included in its own manifest and not as part of the SDK) and re-using the code that way?